viz-lib/src/visualizations/sankey/initSankey.js

```diff
@@ -109,10 +123,33 @@ function spreadNodes(height, data) {
 
 function isDataValid(data) {
   // data should contain column named 'value', otherwise no reason to render anything at all
-  return data && !!find(data.columns, c => c.name === "value");
+  if (!data || !find(data.columns, c => c.name === "value")) {
+    return false;
+  }
+  // prepareData will have coerced any invalid data into NaN, which is accumulated by the reducers below
+  const doAllRowsContainValidData = !isNaN(
+    reduce(data.rows, (finalAcc, row) => finalAcc * reduce(row, (acc, col) => col * acc, 0), 0)
+  );
```

Comment:
```suggestion
  const doAllRowsContainValidData = every(data.rows, row => every(row, v => isFinite(parseFloat(v))));
```

In this case `every` is better than `reduce` because `every` will return on first value that does not match the condition while `reduce` will still process all the data. Also, with `every` you do directly what you mentioned in the comment - check if _every_ value is value, while `reduce` generally serves for a different purpose (fold the data).

P.S. You can even omit that intermediate variable and just return that expression's value from the function.

Reply:
In this case we could actually even coerce the numbers in the very function, but I find keeping them separate clearer and not significant performance-wise